hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -832,15 +833,17 @@ public boolean isPartitionReplicaVersionStale(int partitionId, long[] versions,
 
     @Override
     public Map<Address, List<Integer>> getMemberPartitionsMap() {
-        final Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
-        final int dataMembersSize = dataMembers.size();
+        Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
+        int dataMembersSize = dataMembers.size();
```

Comment:
good one. there should be protection against this. the method should not make assumptions about memberSize > 0. 

@Donnerbart: can you please send a fix.